lib/prometheus/client/push.rb

```diff
@@ -126,6 +133,27 @@ def synchronize
         @mutex.synchronize { yield }
       end
 
+      def validate_no_basic_auth!(uri)
+        if uri.user || uri.password
+          raise ArgumentError, <<~EOF
+            Setting Basic Auth credentials in the gateway URL is not supported, please call the `basic_auth` method
```

Comment:
Driveby 😄 maybe a quick usage example here that a developer could just copy-paste?

Reply:
That's a really good idea. Just added it in!

Also hi Nick 👋🏻